### PR TITLE
General code cleanup

### PR DESCRIPTION
**Pull request by MarcinOrlowski**
General code cleanup incl.
* double assignments removal
* unnecessary checks removal
* code simplification
* merge of duplicate switch() branches

### CODE REVIEW

File: src/main/java/com/cburch/logisim/analyze/model/TruthTable.java [hunk 1]
```diff
@@ -664,11 +664,10 @@ private void removeInput(int index, int oldCount) {
       int b = (1 << (oldCount - 1 - index)); // _0001000
       boolean[] changed = new boolean[columns.size()];
       // loop rows by index to avoid java.util.ConcurrentModificationException
-      for (int i = 0; i < rows.size(); ++i) {
-        Row r = rows.get(i);
-        if (r.inputs[index] == Entry.DONT_CARE)
-          continue;
-        setDontCare(r, b, true, changed); // mutates row
+      for (Row r : rows) {
```
Comment (zdimension) [inline src/main/java/com/cburch/logisim/analyze/model/TruthTable.java:667]: maybe have a look at the comment at line 666

**Review by zdimension — CHANGES_REQUESTED** [review]

File: src/main/java/com/cburch/logisim/analyze/model/TruthTable.java (hunk 1 above)
Comment (MarcinOrlowski) [reply src/main/java/com/cburch/logisim/analyze/model/TruthTable.java:667]: Ough, impressive I missed that mutation. Fixed.

**Comment by maehne** [comment]
Thanks for the clean-up @MarcinOrlowski. Note that your modifications in file `VhdlSyntax.java` are in a risk to be overwritten since it is a file, which is generated by JFlex. The file already contains some clean-ups by @zdimension, which also need to be transferred to the actual source file (cf. to issue #563).

**Comment by MarcinOrlowski** [comment]
That's a trap then. Shouldn't then `VhdlSyntax.java` be NOT checked in at all, but generated at build time? Or at least contain header warning of some sort?

**Comment by maehne** [comment]
Yes, we should let `gradle` handle the generation of `VhdlSyntax.java` and 
not track this generated file with `git`.

**Pull request merged**